src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -17,30 +17,46 @@ import AddSystemButton from '../AddSystemButton/AddSystemButton';
 class DriftTable extends Component {
     constructor(props) {
         super(props);
-        this.systemIds = queryString.parse(this.props.location.search).system_ids;
+        this.setSystemIds();
         this.fetchCompare = this.fetchCompare.bind(this);
+        this.removeSystem = this.removeSystem.bind(this);
         this.formatDate = this.formatDate.bind(this);
     }
 
     async componentDidMount() {
         await window.insights.chrome.auth.getUser();
-        if (this.systemIds) {
-            this.fetchCompare(this.systemIds);
-        }
+        this.fetchCompare(this.systemIds);
+    }
+
+    setSystemIds() {
+        this.systemIds = queryString.parse(this.props.location.search).system_ids;
+        this.systemIds = Array.isArray(this.systemIds) ? this.systemIds : [ this.systemIds ];
+        this.systemIds = this.systemIds.filter(item => item !== undefined);
     }
 
     formatDate(dateString) {
         let date = new Date(dateString);
         return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
     }
 
+    removeSystem(systemId) {
+        let ids = this.systemIds.filter(item => item !== systemId);
+        if (ids.length > 0) {
+            this.fetchCompare(ids);
```

Comment:
Here, we'll need to remove the systemId from the systemIds array. If you:

- Choose 2 systems
- Click one of the Xs (removes as expected)
- Click the X on the other card

You get the other card back that you originally clicked the X on. It's because It's still in the systemIds array.